plugins/services/src/js/utils/ServiceConfigDisplayUtil.js

```diff
@@ -0,0 +1,691 @@
+import classNames from 'classnames';
+import React from 'react';
+import {Table} from 'reactjs-components';
+
+import HostUtil from './HostUtil';
+import Networking from '../../../../../src/js/constants/Networking';
+import Units from '../../../../../src/js/utils/Units';
+import Util from '../../../../../src/js/utils/Util';
+
+const renderMillisecondsFromSeconds = (prop, row) => {
+  let value = row[prop];
+
+  if (value != null) {
+    value = `${value * 1000} ms`;
+  }
+
+  return ServiceConfigDisplayUtil.getDisplayValue(value);
+};
+
+const getColumnClassNameFn = (classes) => {
+  return (prop, sortBy) => {
+    return classNames(classes, {
+      'active': prop === sortBy.prop
+    });
+  };
+};
+
+const getColumnHeadingFn = (defaultHeading) => {
+  return (prop, order, sortBy) => {
+    let caretClassNames = classNames('caret', {
+      [`caret--${order}`]: order != null,
+      'caret--visible': sortBy.prop === prop
+    });
+
+    return (
+      <span>
+        {defaultHeading || prop}
+        <span className={caretClassNames} />
+      </span>
+    );
+  };
+};
+
+const ServiceConfigDisplayUtil = {
+  displayValues: [
+    {
+      values: [
+        {
+          heading: 'General',
+          headingLevel: 1
+        },
+        {
+          key: 'id',
+          label: 'Service ID',
+          transformValue: (value) => {
+            return value.split('/').pop();
+          }
+        },
+        {
+          key: 'instances',
+          label: 'Instances'
+        },
+        {
+          key: 'location',
+          label: 'Location',
+          transformValue: (value, appDefinition) => {
+            let idSegments = Util.findNestedPropertyInObject(appDefinition, 'id').split('/');
+
+            idSegments.pop();
+
+            return `${idSegments.join('/')}/`;
+          }
+        },
+        {
+          key: 'container-runtime', // TODO: Figure out how to find this
+          label: 'Container Runtime'
+        },
+        {
+          key: 'cpus',
+          label: 'CPU'
+        },
+        {
+          key: 'mem',
+          label: 'Memory',
+          transformValue: (value) => {
+            if (value == null) {
+              return ServiceConfigDisplayUtil.getDisplayValue(value);
+            }
+
+            return Units.formatResource('mem', value);
+          }
+        },
+        {
+          key: 'disk',
+          label: 'Disk',
+          transformValue: (value) => {
+            if (value == null) {
+              return ServiceConfigDisplayUtil.getDisplayValue(value);
+            }
+
+            return Units.formatResource('disk', value);
+          }
+        },
+        {
+          key: 'gpus',
+          label: 'GPU'
```

Comment:
Sorry, I'm not sure what you mean by this comment; `gpus` and `cpus` are treated similarly.